streaming/base/format/base/reader.py

```diff
@@ -75,34 +75,50 @@ def __len__(self) -> int:
         """
         return self.samples
 
-    def _evict_raw(self) -> None:
-        """Remove all raw files belonging to this shard."""
+    def _evict_raw(self) -> int:
+        """Remove all raw files belonging to this shard.
+
+        Returns:
+            int: Bytes evicted from cache.
+        """
+        size = 0
         for raw_info, _ in self.file_pairs:
             filename = os.path.join(self.dirname, self.split, raw_info.basename)
             if os.path.exists(filename):
                 os.remove(filename)
+                size += raw_info.bytes
+        return size
 
-    def _evict_zip(self) -> None:
-        """Remove all zip files belonging to this shard."""
+    def _evict_zip(self) -> int:
+        """Remove all zip files belonging to this shard.
+
+        Returns:
+            int: Bytes evicted from cache.
+        """
+        size = 0
         for _, zip_info in self.file_pairs:
             if zip_info:
                 filename = os.path.join(self.dirname, self.split, zip_info.basename)
                 if os.path.exists(filename):
                     os.remove(filename)
+                    size += zip_info.bytes
+        return size
 
-    def evict(self) -> None:
-        """Remove all files belonging to this shard."""
-        self._evict_raw()
-        self._evict_zip()
+    def evict(self) -> int:
+        """Remove all files belonging to this shard.
 
-    def init_local_dir(self, filenames_present: Set[str], keep_zip: bool) -> bool:
+        Returns:
+            int: Bytes evicted from cache.
+        """
+        return self._evict_raw() + self._evict_zip()
+
+    def set_up_local(self, listing: Set[str], safe_keep_zip: bool) -> int:
         """Bring what shard files are present to a consistent state, returning whether present.
 
         Args:
-            filenames_present (Set[str]): The listing of all files under dirname/[split/]. This is
-                listed once and then saved because there could potentially be very many shard
-                files.
-            keep_zip (bool): Whether to keep zip files when decompressing. Possible when
+            listing (Set[str]): The listing of all files under dirname/[split/]. This is listed
+                once and then saved because there could potentially be very many shard files.
+            safe_keep_zip (bool): Whether to keep zip files when decompressing. Possible when
```

Comment:
isn't `keep_zip` more intuitive ?

Reply:
`safe_keep_zip` is the name of a field in Stream which is derived from Stream arg `keep_zip` but protecting against blowing away the shards when your local is the remote/authoritative source